### PR TITLE
Fix: make TaggedData data field optional

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Without this change it fails to parse payloads where the data field is optional
```
dacite.exceptions.UnionMatchError: can not match type "dict" to any type of "payload" union: typing.Union[iota_sdk.types.payload.TaggedDataPayload, iota_sdk.types.payload.TransactionPayload, iota_sdk.types.payload.MilestonePayload, NoneType]
```

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

```python
from iota_sdk import Client, HexStr, Utils, OutputType

client = Client(nodes=['https://api.testnet.shimmer.network'])

transaction_id = HexStr(
    "0xe08103c471a1ccbd65b248a1bf0929b9b9c5e12bb219f3971aadf95b870a1ed4")
pub_key_hash = Utils.bech32_to_hex(
    "rms1qpe9sae0jc5ra0xfrxtcqkstftxf3jpa859mr0y3nccewjake656yu2k3pk")
amount = 1_000_000_000


block = client.get_included_block(transaction_id)


for output in block.payload.essence.outputs:
    if output.type == OutputType.Basic:
        if len(output.unlockConditions) == 1 and output.unlockConditions[0].type == 0:
            if output.unlockConditions[0].address.pubKeyHash == pub_key_hash and int(output.amount) == amount:
                print(f"{pub_key_hash} got {int(amount)}")

```

### CODE REVIEW

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/python/iota_sdk/types/payload.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ class TaggedDataPayload(Payload):
         data: The data part of the tagged data payload.
     """
     tag: HexStr
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/payload.py]: Can also be Optional, no?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/payload.py]: True

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/payload.py]: Somehow this fails now, will not fix it today anymore and update the dates later

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]
conflicts :trollface:

File: bindings/python/iota_sdk/types/payload.py (hunk 1 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/payload.py]: Optional now

**Comment by Thoralf-M** [comment]
Draft for now, maybe we can find a better way (without requiring the type in the TaggedDataPayload __init__)

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**